aws-api-appsync/src/main/java/com/amplifyframework/api/aws/AuthorizationType.java

```diff
@@ -71,5 +75,30 @@ public static AuthorizationType from(String name) {
 
         throw new IllegalArgumentException("No such authorization type: " + name);
     }
+
+    /**
+     * Look up an AuthorizationType by inspecting an AuthRule annotation.
+     * @param authRuleAnnotation The annotation obtained from a model.
+     * @return The AuthorizationType for the provider
```

Comment:
```suggestion
     * @return The AuthorizationType for the provider
     * @throws IllegalArgumentException if AuthRule's provider does not match an authorization type
```
We should document this behavior since it's not clear what this method returns in such case.